### PR TITLE
Add test for duplicate EM_ASM strings

### PR DESCRIPTION
**Pull request by sbc100**
See https://github.com/WebAssembly/binaryen/pull/2422

### CODE REVIEW

**Comment by sbc100** [comment]
This test should fail until the binaryen fix is rolled in, at which point we should be able to land this.

File: tests/test_other.py [hunk 1]
```diff
@@ -9780,6 +9780,34 @@ def test_assertions_on_outgoing_module_api_changes(self):
 Module.arguments has been replaced with plain arguments_
 ''', run_js('a.out.js', assert_returncode=None, stderr=PIPE))
 
+  def test_em_asm_duplicate_strings(self):
+    # We had a regression where tow different EM_ASM strings from two diffferent
```
Comment (jgravelle-google) [inline tests/test_other.py]: Typo: s/tow/two/

Should we link to the issue here? Should add even more context

File: tests/test_other.py [hunk 2]
```diff
@@ -9780,6 +9780,34 @@ def test_assertions_on_outgoing_module_api_changes(self):
...
+    # object files were de-duplicated in wasm-emscripten-finalize.  This used to
+    # work when we used zero-based index for store the JS strings, but one we
```
Comment (jgravelle-google) [inline tests/test_other.py]: s/one/once/

File: tests/test_other.py (hunk 1 above)
Comment (sbc100) [reply tests/test_other.py]: Err, these is no issue :(
